### PR TITLE
HypergraphConv TorchScript Compilation Compatibility

### PR DESCRIPTION
**Pull request by chrisrosner**
Addresses #10399

This was the minimal set of changes I could figure out to get HypergraphConv to compile with torchscript.

Included are unit tests demonstrating the problems.

The following issues were resolved:
type of alpha changing from None -> tensor
torchscript not understanding the import functional as F
non-attention branch does not define some member variables
default value for dropout is wrong type

## Test Plan
pytest -k hyper

### CODE REVIEW

File: test/nn/conv/test_hypergraph_conv.py [hunk 1]
```diff
@@ -46,3 +46,23 @@ def test_hypergraph_conv_with_more_edges_than_nodes():
...
+    in_channels, out_channels = (2, 3)
+    conv = HypergraphConv(in_channels, out_channels)
+    script = torch.jit.script(conv)
+    output = script(torch.randn(4, in_channels),
+                    torch.tensor([[0, 1, 2], [0, 0, 1]]),
+                    hyperedge_attr=torch.randn(2, in_channels))
+    assert output.size() == (4, out_channels)
+
+
+def test_hypergraph_jit_with_attention():
```
Comment (wsad1) [inline test/nn/conv/test_hypergraph_conv.py]:
```suggestion
```

**Comment by wsad1** [comment]
@chrisrosner thanks for the work. 
Could you enable ` Maintainers are allowed to edit this pull request.` for this PR. Need to merge master so that tests pass.

File: test/nn/conv/test_hypergraph_conv.py [hunk 2]
```diff
@@ -46,3 +46,13 @@ def test_hypergraph_conv_with_more_edges_than_nodes():
...
+
+
+def test_hypergraph_jit():
+    in_channels, out_channels = (2, 3)
+    conv = HypergraphConv(in_channels, out_channels, use_attention=True)
+    script = torch.jit.script(conv)
+    output = script(torch.randn(4, in_channels),
+                    torch.tensor([[0, 1, 2], [0, 0, 1]]),
+                    hyperedge_attr=torch.randn(2, in_channels))
+    assert output.size() == (4, out_channels * conv.heads)
```
Comment (wsad1) [inline test/nn/conv/test_hypergraph_conv.py:58]:
```suggestion
    x = torch.randn(4, in_channels)
    out_1 = conv(x, torch.tensor([[0, 1, 2], [0, 0, 1]]),
                    hyperedge_attr=torch.randn(2, in_channels))
    out_2 = script(x,
                    torch.tensor([[0, 1, 2], [0, 0, 1]]),
                    hyperedge_attr=torch.randn(2, in_channels))
    assert torch.allclose(out_1, out_2)
```

Also add the `onlyFullTest` decorator like https://github.com/pyg-team/pytorch_geometric/blob/b2a1fdcf0d8f7622e00fec979b3e45eb384870c6/test/nn/pool/test_pool.py#L10. 
Good to merge after this.

**Comment by akihironitta** [comment]
Closing this PR for now since we don't have permission to push any change to this branch to get it ready for merging. Feel free to re-open and let us know if you're still interested in working on this. Thank you for your contribution!
